docs/extend/backend-events.md

```diff
@@ -0,0 +1,125 @@
+Historically, Flarum has used events for its extension API, emitting events like `GetDisplayName` or `ConfigureApiRoutes` to allow extensions to insert logic into various parts of Flarum. These events are gradually being phased out in favor of the declarative [extender system](start.md#extenders), and will be removed before stable. Domain events will not be removed.
+:::
+
+For a full list of backend events, see our [API documentation](https://api.docs.flarum.org/php/master/search.html?search=Event). Domain events classes are organized by namespace, usually `Flarum\TYPE\Event`.
+
+## Listening to Events
+
+You can attach a listener to an event using the [`Event`](https://api.docs.flarum.org/php/master/flarum/extend/event) [extender](start.md#extenders):
+
+```php
+use Flarum\Extend;
+use Flarum\Post\Event\Deleted;
+use Symfony\Component\Translation\TranslatorInterface;
+
+
+return [
+    (new Extend\Event)
+        ->listen(Deleted::class, function($event) {
+          // do something here
+        })
+        ->listen(Deleted::class, PostDeletedListener::class)
+];
+
+
+class PostDeletedListener
+{
+  protected $translator;
+
+  public function __construct(TranslatorInterface $translator)
+  {
+      $this->translator = $translator;
+  }
+
+  public function handle(Deleted $event)
+  {
+    // Your logic here
+  }
+}
+```
+
+As shown above, a listener class can be used instead of a callback. This allows you to [inject dependencies](https://laravel.com/docs/6.x/container) into your listener class.
+
+## Dispatching Events
+
+Dispatching events is very simple. All you need to do is inject `Illuminate\Contracts\Events\Dispatcher` into your class, and then call its `dispatch` method. For instance:
+
+```php
+use Flarum\Post\Event\Deleted;
+use Illuminate\Contracts\Events\Dispatcher;
+
+
+class SomeClass
+{
+  /**
+    * @var Dispatcher
+    */
+  protected $events;
+
+  /**
+   * @param Dispatcher $events
+   */
+  public function __construct(Dispatcher $events)
+  {
+    $this->events = $events;
+  }
+
+  public function someMethod()
+  {
+    // Logic
+    $this->events->dispatch(
+      new Deleted($somePost, $someActor)
+    );
+    // More Logic
+  }
+}
+```
+
+## Custom Events
+
+As an extension developer you can define your own events to allow yourself (or other extensions) to react to events in your extension.
+Events are generally just POPOs (Plain Old PHP Objects). When defining a new event, you'll typically want to use public properties, and maybe some methods for convenience of users.
+For example, if we take a look at `Flarum\Post\Event\Deleted`, it's just a wrapping around some data:
+
+```php
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Post\Event;
+
+use Flarum\Post\Post;
+use Flarum\User\User;
+
+class Deleted
+{
+    /**
+     * @var \Flarum\Post\Post
```

Comment:
I'd suggest dropping the docblocks to shorten the example.

If we keep the docblocks, they are highly inconsistent. This one uses an unnecessarily full path, and the docblock for the constructor is missing one argument.

Reply:
It's copy-pasted from core :grimacing: